From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Tue, 19 Jan 2016 23:18:56 +0100
Subject: [PATCH] {BUG} amd: powerplay: do not return uninitialized value

clang reports:
  CC [M]  drivers/gpu/drm/amd/amdgpu/../powerplay/hwmgr/processpptables.o
drivers/gpu/drm/amd/amdgpu/../powerplay/hwmgr/processpptables.c:1014:11: error: variable 'result' is used
      uninitialized whenever 'if' condition is false [-Werror,-Wsometimes-uninitialized]
        else if ((fw_info->ucTableFormatRevision == 2)
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
drivers/gpu/drm/amd/amdgpu/../powerplay/hwmgr/processpptables.c:1027:9: note: uninitialized use occurs here
        return result;
               ^~~~~~
drivers/gpu/drm/amd/amdgpu/../powerplay/hwmgr/processpptables.c:1014:7: note: remove the 'if' if its condition is
      always true
        else if ((fw_info->ucTableFormatRevision == 2)
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
drivers/gpu/drm/amd/amdgpu/../powerplay/hwmgr/processpptables.c:1014:11: error: variable 'result' is used
      uninitialized whenever '&&' condition is false [-Werror,-Wsometimes-uninitialized]
        else if ((fw_info->ucTableFormatRevision == 2)
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
drivers/gpu/drm/amd/amdgpu/../powerplay/hwmgr/processpptables.c:1027:9: note: uninitialized use occurs here
        return result;
               ^~~~~~
drivers/gpu/drm/amd/amdgpu/../powerplay/hwmgr/processpptables.c:1014:11: note: remove the '&&' if its condition is
      always true
        else if ((fw_info->ucTableFormatRevision == 2)
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
drivers/gpu/drm/amd/amdgpu/../powerplay/hwmgr/processpptables.c:992:12: note: initialize the variable 'result' to
      silence this warning
        int result;
                  ^
                   = 0
2 errors generated.
---
 drivers/gpu/drm/amd/powerplay/hwmgr/processpptables.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/powerplay/hwmgr/processpptables.c b/drivers/gpu/drm/amd/powerplay/hwmgr/processpptables.c
index b64f42d2c3ae..0229ac636c4e 100644
--- a/drivers/gpu/drm/amd/powerplay/hwmgr/processpptables.c
+++ b/drivers/gpu/drm/amd/powerplay/hwmgr/processpptables.c
@@ -989,7 +989,7 @@ static int init_overdrive_limits_V2_1(struct pp_hwmgr *hwmgr,
 static int init_overdrive_limits(struct pp_hwmgr *hwmgr,
 			const ATOM_PPLIB_POWERPLAYTABLE *powerplay_table)
 {
-	int result;
+	int result = 0 /* BUG */;
 	uint8_t frev, crev;
 	uint16_t size;
 
-- 
